### PR TITLE
resolve lock the workbench resources when using GradleLaunchConfiguration to run task with "--continuous"

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiJob.java [hunk 1]
```diff
@@ -91,7 +97,7 @@ protected void canceling() {
     /**
      * Default handler for the target operation.
      *
-     * @param <T> the result type
+     * @param <T> the result typeToolingApiJobToolingApiJob
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiJob.java]: This looks like an accidental change.

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperation.java [hunk 2]
```diff
@@ -28,6 +29,14 @@
      */
     public String getName();
 
+    /**
+     * Returns the operation Scheduling rule.
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperation.java]: > Returns the scheduling rule that is applied when the operation is executed synchronously or asynchronously.

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperation.java [hunk 3]
```diff
@@ -28,6 +29,14 @@
...
+     *
+     * @return the operation Scheduling rule
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperation.java]: ` @return the operation's scheduling rule`

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperations.java [hunk 4]
```diff
@@ -31,7 +33,7 @@ private ToolingApiOperations() {
      * If the first operation is cancelled then the second operation is not executed.
      *
      * @param op1 the operation to execute first
-     * @param op2 the operation to execute last
+     * @param op2 the operation to execute last        // TODO Auto-generated method stub
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperations.java]: This looks like an accidental change.

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperations.java [hunk 5]
```diff
@@ -56,6 +58,11 @@ public void runInToolingApi(CancellationTokenSource tokenSource, IProgressMonito
 
                 op2.runInToolingApi(tokenSource, progress.newChild(1));
             }
+
+            @Override
+            public ISchedulingRule getRule() {
+                return ResourcesPlugin.getWorkspace().getRoot();
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/ToolingApiOperations.java]: `MultiRule` should be returned.

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/internal/DefaultToolingApiOperationManager.java [hunk 6]
```diff
@@ -38,7 +39,7 @@ public void run(ToolingApiOperation runnable, IProgressMonitor monitor) throws C
     @Override
     public void run(ToolingApiOperation runnable, CancellationTokenSource tokenSource, IProgressMonitor monitor) throws CoreException {
         IProgressMonitor efficientMonitor = new RateLimitingProgressMonitor(monitor, 500, TimeUnit.MILLISECONDS);
-        ResourcesPlugin.getWorkspace().run(new WorkspaceRunnableAdapter(runnable, tokenSource), efficientMonitor);
+        ResourcesPlugin.getWorkspace().run(new WorkspaceRunnableAdapter(runnable, tokenSource), runnable.getRule(), IWorkspace.AVOID_UPDATE, efficientMonitor);
```
Comment (donat) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/operation/internal/DefaultToolingApiOperationManager.java]: Can you elaborate explain why the `AVOID_WORKSPACE` is necessary here?

**Comment by simonjhy** [comment]
After change I should push a new commit or still use same commit?

**Comment by simonjhy** [comment]
hi, donat. I added an integration test in RunGradleBuildLaunchRequestJobTest. I think it isn't need to add UI SWT test to check this fix. Is this ok?

**Comment by donat** [comment]
> After change I should push a new commit or still use same commit?

It's up to you, the GH reviews can handle both. Personally, I reuse the commits and use force-push to rewrite remote history when I'm working on a feature branch and I want to avoid having lots of small commits.

**Comment by donat** [comment]
> I think it isn't need to add UI SWT test to check this fix. Is this ok?

**Comment by simonjhy** [comment]
ok, I will merge two commits into one and sent again.

**Comment by donat** [comment]
PR merged. Thank you for the contribution!

**Comment by simonjhy** [comment]
Thanks you, happy to see it merged

**Pull request merged**